### ISSUE TITLE
Font preview doesn't work correctly

### ISSUE THREAD

**Issue by fflorent** [firebug/firebug.next#257]
STR:
1. Go to http://nightly.mozilla.org
2. Open Firebug, inspect #nightly-title
3. Open the "Font" side-panel.

Here is what it looks like currently:

![font_sidepanel](https://cloud.githubusercontent.com/assets/371705/5603020/91988e2a-9371-11e4-813a-95372035f58c.png)

Florent

**Comment by anurag90x** [comment]
Hmm, there's some weird behaviour here. If Firebug is just turned on, then we can't really see anything but BrowserToolbox shows an img there. So I switched to the light theme and saw this -

![screen shot 2015-01-03 at 11 34 36 pm](https://cloud.githubusercontent.com/assets/1813816/5603240/3be90074-93a1-11e4-8f2a-8cf7197b828c.png)

Note that the font preview is not the usual color for the light theme (usually black). So now if I just close Firebug (while in light theme with this weird color scheme for the preview) and reopen it we get the usual color scheme for the preview (black) and then switching to Firebug theme produces -


![screen shot 2015-01-03 at 11 33 18 pm](https://cloud.githubusercontent.com/assets/1813816/5603244/9891e610-93a1-11e4-977a-ff9934df947e.png)

**Comment by anurag90x** [comment]
So, I think I found the offending line of code but it's inside font-inspector.js in devtools -

```javascript
    let fillStyle = (Services.prefs.getCharPref("devtools.theme") == "light") ?
        "black" : "white";
    let options = {
      includePreviews: true,
      previewFillStyle: fillStyle
    }
```
So, when the request goes to the pageStyle actor it sees the previewStyle as white. This explains the weird behaviour above. Basically, by defaulting to Firebug theme we get the white preview style then if we go to Light theme we are still seeing white. But if we change the node selection in the light theme, the preview style is black again (same effect obtained by closing and reopening firebug in light theme). Now, if we switch to Firebug theme we see the black preview style but changing the node again goes back to white ( because of the theme check outlined above).

Any thoughts, @fflorent ?

**Comment by janodvarko** [comment]
@anurag90x great analyses!

I created a bugzilla report for this:
https://bugzilla.mozilla.org/show_bug.cgi?id=1118179

Honza

**Comment by fflorent** [comment]
I can confirm it works for me. Thanks @anurag90x and @janodvarko.

Florent

**Comment by fflorent** [comment]
(Sorry, closed by mistake)

**Comment by janodvarko** [comment]
Assigning myself. I just created a patch that should fix the issue on the platform.
https://bugzilla.mozilla.org/show_bug.cgi?id=1118179
(it's rather a workaround at this point)

As soon as the patch lands I'll remove the workaround on Firebug side.

Honza

**Comment by janodvarko** [comment]
Fixed on the platform, closing.

Honza

**Comment by janodvarko** [comment]
$fixed firebug-3.0.0-alpha.7
